lib/index.js

```diff
@@ -4,6 +4,11 @@
  */
 import KEYS from "./visitor-keys.js";
 
+/**
+ * @typedef {{ [type: string]: ReadonlyArray<string> }} VisitorKeysWritable
```

Comment:
Can we somehow _not_ export `VisitorKeysWritable`? It's only used internally, so it might be confusing why are we exporting a type that isn't used in the API.

Reply:
This issue might be related, but it seems there's no answer yet: https://github.com/microsoft/TypeScript/issues/46011

Reply:
Yes we can (at the small cost of our unit tests needing to redefine the type which is fine).

@nzakas had asked about using `@typedef` for  shorter type usage throughout the document, but yes, it seems like due to [this issue](https://github.com/microsoft/TypeScript/issues/46011) indicating we cannot control typedef exporting, using a typedef at this time requires us to make the type exported. I would think therefore that it a good idea to avoid using typedefs for local types at this time.

We could at least use newlines to make the type more readable when inline, e.g.,

```js
const retv = /** @type {{
    [type: string]: ReadonlyArray<string>
}} */ (Object.assign({}, KEYS));
```

Reply:
Yes, if there's currently no way to tell that some types are internal, I think it's better to inline types we don't want to export until that issue gets resolved.

Reply:
Ok, I've added a commit to avoid exporting that type. I did redefine the type within the testing file for convenience and clarity, but that testing file isn't being exported (and the type isn't defined that way anyways).